designs/lily/src/front.mjs

```diff
@@ -0,0 +1,586 @@
+    // leggings must reach to fork at least, so define a minimum
+    let waistToFork = points.waistX.dy(points.fork)
+    let waistToKnee = points.waistX.dy(points.knee) // adapting the seams may have shifted the knee up or down
+    if (waistToFork > requestedLength) {
+      //log.warning('length reduction capped; cutting off at fork') // log only for back part
+      // add one percent to waistToFork to ensure that path length is nonzero
+      requestedLength = waistToFork * 1.01
+    }
+
+    // work-around to avoid splitting exactly at the knee
+    //   (due to a bug, splitting a path at a node is not possible)
+    if (0.999 < requestedLength / waistToKnee && requestedLength / waistToKnee < 1.001) {
+      requestedLength = 1.001 * waistToKnee
+    }
+    points.bottom = points.waistX.shift(270, requestedLength)
+    let upperPoint, upperCp
+    if (requestedLength < waistToKnee) {
+      extendBeyondKnee = 0
+
+      // 'cut' between fork and knee
+      if (points.waistOut.x < points.seatOut.x) {
+        upperPoint = points.styleWaistOutLily
+        upperCp = points.seatOut
+      } else {
+        upperPoint = points.seatOut
+        upperCp = points.seatOutCp2
+      }
+      points.bottomOut = utils.lineIntersectsCurve(
+        points.bottom.shift(0, 999),
+        points.bottom.shift(180, 999),
+        points.kneeOut,
+        points.kneeOutCp1,
+        upperCp,
+        upperPoint
+      )
+
+      points.bottomIn = utils.lineIntersectsCurve(
+        points.bottom.shift(0, 999),
+        points.bottom.shift(180, 999),
+        points.kneeIn,
+        points.kneeInCp2,
+        points.forkCp1,
+        points.fork
+      )
+    } else {
+      // 'cut' between knee and 'floor'
+      points.bottomOut = utils.lineIntersectsCurve(
+        points.bottom.shift(0, 999),
+        points.bottom.shift(180, 999),
+        points.kneeOut,
+        points.kneeOutCp2,
+        points.floorOutCp2,
+        points.floorOut
+      )
+
+      points.bottomIn = utils.lineIntersectsCurve(
+        points.bottom.shift(0, 999),
+        points.bottom.shift(180, 999),
+        points.kneeIn,
+        points.kneeInCp1,
+        points.floorInCp2,
+        points.floorIn
+      )
+    }
+
+    // define the three parts of the path, then combine
+    paths.bottom = new Path().move(points.bottomOut).line(points.bottomIn)
+
+    const halves = paths.seam.split(points.bottomOut)
+    paths.upperOutseam = halves[0]
+    const halves2 = halves[1].split(points.bottomIn)
+    paths.upperInseam = halves2[1]
+
+    paths.seam = paths.upperOutseam.join(paths.bottom).join(paths.upperInseam)
+  } else {
+    // define the same three parts of the path as when length reduction is enabled, then combine
+
+    // first define the points (also used for paperless)
+    points.bottom = points.floor
+    points.bottomIn = points.floorIn
+    points.bottomOut = points.floorOut
+
+    paths.bottom = new Path().move(points.bottomIn).line(points.bottomOut)
```

Comment:
```suggestion
    paths.bottom = new Path().move(points.bottomOut).line(points.bottomIn)
```
The hem allowance is inverted when Length Reduction is set to default:
![Screenshot 2024-03-16 at 6 29 34 PM](https://github.com/freesewing/freesewing/assets/109869956/8dd1bdf1-0a89-4ce4-9c1a-35a3542d0e34)